crypto/s2n_openssl.h

```diff
@@ -17,28 +17,26 @@
 
 #include <stdbool.h>
 
-/**
- * openssl with OPENSSL_VERSION_NUMBER < 0x10100003L made data type details unavailable
- * libressl use openssl with data type details available, but mandatorily set
- * OPENSSL_VERSION_NUMBER = 0x20000000L, insane!
- * https://github.com/aws/aws-sdk-cpp/pull/507/commits/2c99f1fe0c4b4683280caeb161538d4724d6a179
- */
-#if defined(LIBRESSL_VERSION_NUMBER) && (OPENSSL_VERSION_NUMBER == 0x20000000L)
-#undef OPENSSL_VERSION_NUMBER
-#define OPENSSL_VERSION_NUMBER 0x1000107fL
-#endif
```

Comment:
I would prefer not to remove this comment or logic so it's clear why things are the way they are. Also, rather than modifying the `S2N_OPENSSL_VERSION_AT_LEAST` macro, why not just set the inferred API version here?

```c
/* constructs a openssl version number */
#define S2N_FORMAT_OPENSSL_VERSION(major, minor, fix) ((major << 28) + (minor << 20) + (fix << 12))

/**
 * openssl with OPENSSL_VERSION_NUMBER < 0x10100003L made data type details unavailable
 * libressl use openssl with data type details available, but mandatorily set
 * OPENSSL_VERSION_NUMBER = 0x20000000L, insane!
 * https://github.com/aws/aws-sdk-cpp/pull/507/commits/2c99f1fe0c4b4683280caeb161538d4724d6a179
 */
#if defined(LIBRESSL_VERSION_NUMBER)
#  undef OPENSSL_VERSION_NUMBER

/**
 * >=3.5 supports the libcrypto 1.1 API; otherwise use 1.0.
 * see: https://ftp.openbsd.org/pub/OpenBSD/LibreSSL/libressl-3.5.0-relnotes.txt
 */
#  if (LIBRESSL_VERSION_NUMBER >= S2N_FORMAT_OPENSSL_VERSION(3, 5, 0))
#    define OPENSSL_VERSION_NUMBER S2N_FORMAT_OPENSSL_VERSION(1, 1, 1)
#  else
#    define OPENSSL_VERSION_NUMBER S2N_FORMAT_OPENSSL_VERSION(1, 0, 1)
#  endif

#endif
```

Reply:
ah, yes. sorry. This is flawed anyway. I will fix this in both patches